### PR TITLE
Fix all Rails' warnings

### LINKED ISSUE #1349: DEPRECATION WARNING: [] is deprecated and will be removed from Rails 6.2 (Use configs_for)
## Description

On application startup the following deprecation warning occurs:
```
ActiveSupport::DeprecationException:
  DEPRECATION WARNING: [] is deprecated and will be removed from Rails 6.2 (Use configs_for) (called from block in <top (required)> at my-app/config/initializers/datadog.rb:9)
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/vendor/active_record/connection_specification.rb:269:in `resolve_symbol_connection'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/vendor/active_record/connection_specification.rb:253:in `resolve_connection'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/vendor/active_record/connection_specification.rb:155:in `resolve'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/contrib/active_record/configuration/resolver.rb:15:in `resolve'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/contrib/configurable.rb:34:in `configuration_for?'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/contrib/configurable.rb:49:in `configure'
# gem_path/ruby-2.7.2/gems/ddtrace-0.45.0/lib/ddtrace/contrib/extensions.rb:80:in `instrument'
# ./config/initializers/datadog.rb:9:in `block in <top (required)>'
```

The warning is triggered by this line extracted from my  `./config/initializers/datadog.rb`
```
  c.use :active_record, describes: :some_database, service_name: 'some-service-name'
```

## Versions

ddtrace (0.45.0)
rails 6.1.1

## Desired result

No deprecation warnings are emitted.

## Related information

Rails commit that adds the deprecation: https://github.com/rails/rails/commit/2a53fe638de2c485c2655b6b7236ec1ab7dca5ad

### PR DESCRIPTION
**Pull request by marcotc**
Fixes #1349

This PR fixes current Rails warnings triggered by the tracer and creates a check for all Rails tests to ensure deprecation warnings are never emitted Rails.

### CODE REVIEW

File: spec/ddtrace/contrib/action_cable/instrumentation_spec.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 require 'ddtrace/contrib/support/spec_helper'
+require 'spec/ddtrace/contrib/rails/support/deprecation.rb'
```
Comment (delner) [inline spec/ddtrace/contrib/action_cable/instrumentation_spec.rb:2]: Is the `.rb` necessary?

File: spec/ddtrace/contrib/rails/support/deprecation.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Raises error when a deprecation would be emitted
+def raise_on_rails_deprecation!
+  # DEV: In Rails 6.1 `ActiveSupport::Deprecation.disallowed_warnings`
+  # DEV: was introduced that allows fine grain configuration
```
Comment (delner) [inline spec/ddtrace/contrib/rails/support/deprecation.rb:4]: I don't think we need a `DEV` per line; just `DEV` on the first line and matching indentation should be sufficient.

**Review by delner — APPROVED** [review]
Minor comments but otherwise looks good.

File: spec/ddtrace/contrib/action_cable/instrumentation_spec.rb (hunk 1 above)
Comment (marcotc) [reply spec/ddtrace/contrib/action_cable/instrumentation_spec.rb]: No need!

File: spec/ddtrace/contrib/rails/support/deprecation.rb (hunk 2 above)
Comment (marcotc) [reply spec/ddtrace/contrib/rails/support/deprecation.rb]: Done!

**Comment by marcotc** [comment]
Ready for another round of review after last comments.

**Review by delner — APPROVED** [review]
:+1:

**Pull request merged**